scripts.js

```diff
@@ -0,0 +1,327 @@
+
+const numbers = [1, 2, 3];
+const squaredNumbers = getSquaredNumbers(numbers);
+console.log(squaredNumbers); // [1, 4, 9]
+console.log(numbers); // [1, 2, 3]
+console.log(numbers !== squaredNumbers); // true
+
+// 4
+function getReversedString(string) {
+  let characters = string.split('');
+  characters.reverse();
+
+  return characters.join('');
+}
+
+console.log(getReversedString('Hello!')); // '!olleH'
+console.log(getReversedString('Arrays')); // 'syarrA'
+
+//5
+function isPalindrome(string) {
+  string = string.replace(/\s/g, '');
+
+  return string.toLowerCase() === getReversedString(string.toLowerCase());
+}
+
+console.log(isPalindrome('Kayak')); // true
+console.log(isPalindrome('Racecar')); // true
+console.log(isPalindrome('Was it a cat I saw')); // true
+console.log(isPalindrome('Hello!')); // false
+
+//6
+function countLetters(string) {
+  let countedLetters= {};
+  let characters = string.replace(/\s/g, '');
+  characters = characters.toLowerCase();
+
+  for (let i = 0; i < characters.length; ++i){
+    let character = characters.charAt(i)
+    if (!countedLetters[character]) {
+      countedLetters[character] = 1;
+    }
+    else {
+      countedLetters[character] += 1;
+    }
+  }
+
+  return countedLetters;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7
+function getYoungestPerson(peopleArray) {
+  let youngestPersonAge = peopleArray[0].age;
+
+  for (let i = 0; i < peopleArray.length; ++i) {
+    if (youngestPersonAge > peopleArray[i].age) {
+      youngestPersonAge = peopleArray[i].age;
+    }
+  }
+
+  return youngestPersonAge;
+}
+
+function getOldestPerson(peopleArray) {
+  let oldestPersonAge = 0;
+
+  for (let i = 0; i < peopleArray.length; ++i) {
+    if (oldestPersonAge < peopleArray[i].age) {
+      oldestPersonAge = peopleArray[i].age;
+    }
+  }
+
+  return oldestPersonAge;
+}
+
+function getAgeDifference(peopleArray) {
+  return getOldestPerson(peopleArray) - getYoungestPerson(peopleArray);
+}
+
+const peopleArray = [
+  {
+    name: 'Adam',
+    age: 20
+  },
+  {
+    name: 'Amanda',
+    age: 5
+  },
+  {
+    name: 'John',
+    age: 75
+  },
+  {
+    name: 'Dave',
+    age: 15
+  }
+]
+
+console.log(getYoungestPerson(peopleArray));
+console.log(getOldestPerson(peopleArray));
+console.log(getAgeDifference(peopleArray)); // 70
+
+// 8
+function positiveSum(numbers) {
+  let numbersSum = 0;
+
+  for (let i = 0; i < numbers.length; ++i) {
+    if (numbers[i] > 0) {
+      numbersSum = numbersSum + numbers[i];
+    }
+  }
+
+  return numbersSum;
+}
+
+console.log(positiveSum([1, 5, 10]));
+console.log(positiveSum([-1, -5, -10]));
+
+// 9
+function squaredNumbersSummary(numbers) {
+  let total = 0;
+
+  for (let i = 0; i < numbers.length; ++i) {
+    total = total + numbers[i] * numbers[i];
+  }
+
+  return total;
+}
+
+console.log(squaredNumbersSummary([0, 3, 4, 5]));
+
+// 10
+function abbreviationName(name) {
```

Comment:
`abbreviation` is a noun. Since a function is supposed to _do_ something, use a verb instead.